statsmodels/stats/contingency_tables.py

```diff
@@ -0,0 +1,1312 @@
+            The Z-score for the test statistic.
+        pvalue : float
+            The p-value for the test.
+
+        Notes
+        -----
+        The scores define the trend to which the test is most sensitive.
+
+        Using the default row and column scores gives the
+        Cochran-Armitage trend test.
+        """
+
+        if row_scores is None:
+            row_scores = np.arange(self.table.shape[0])
+
+        if col_scores is None:
+            col_scores = np.arange(self.table.shape[1])
+
+        if len(row_scores) != self.table.shape[0]:
+            raise ValueError("The length of `row_scores` must match the first dimension of `table`.")
+
+        if len(col_scores) != self.table.shape[1]:
+            raise ValueError("The length of `col_scores` must match the second dimension of `table`.")
+
+        # The test statistic
+        statistic = np.dot(row_scores, np.dot(self.table, col_scores))
+
+        # Some needed quantities
+        n_obs = self.table.sum()
+        rtot = self.table.sum(1)
+        um = np.dot(row_scores, rtot)
+        u2m = np.dot(row_scores**2, rtot)
+        ctot = self.table.sum(0)
+        vn = np.dot(col_scores, ctot)
+        v2n = np.dot(col_scores**2, ctot)
+
+        # The null mean and variance of the test statistic
+        e_stat = um * vn / n_obs
+        v_stat = (u2m - um**2 / n_obs) * (v2n - vn**2 / n_obs) / (n_obs - 1)
+        sd_stat = np.sqrt(v_stat)
+
+        zscore = (statistic - e_stat) / sd_stat
+        pvalue = 2 * stats.norm.cdf(-np.abs(zscore))
+
+        b = _Bunch()
+        b.statistic = statistic
+        b.null_mean = e_stat
+        b.null_sd = sd_stat
+        b.zscore = zscore
+        b.pvalue = pvalue
+        return b
+
+
+    @cache_readonly
+    def marginal_probabilities(self):
+        # docstring for cached attributes in init above
+
+        n = self.table.sum()
+        row = self.table.sum(1) / n
+        col = self.table.sum(0) / n
+
+        if isinstance(self.table_orig, pd.DataFrame):
+            row = pd.Series(row, self.table_orig.index)
+            col = pd.Series(col, self.table_orig.columns)
+
+        return row, col
+
+
+    @cache_readonly
+    def independence_probabilities(self):
+        # docstring for cached attributes in init above
+
+        row, col = self.marginal_probabilities
+        itab = np.outer(row, col)
+
+        if isinstance(self.table_orig, pd.DataFrame):
+            itab = pd.DataFrame(itab, self.table_orig.index,
+                                self.table_orig.columns)
+
+        return itab
+
+
+    @cache_readonly
+    def fittedvalues(self):
+        # docstring for cached attributes in init above
+
+        probs = self.independence_probabilities
+        fit = self.table.sum() * probs
+        return fit
+
+
+    @cache_readonly
+    def pearson_resids(self):
+        # docstring for cached attributes in init above
+
+        fit = self.fittedvalues
+        resids = (self.table - fit) / np.sqrt(fit)
```

Comment:
The Pearson residuals use the Poisson model for standardization.  I'm using this definition:

https://onlinecourses.science.psu.edu/stat504/node/86